File: src/main/java/org/example/HomeWork/hw3/DrinksComparator.java
package org.example.HomeWork.hw3;

import java.util.Comparator;


public class DrinksComparator<Drinks> implements Comparator<Drinkss> {
    private String type;

    public DrinksComparator(String type) {
        this.type = type;
    }

    @Override
    public int compare(Drinkss o1, Drinkss o2) {
        switch (type) {
            case "volume": {
                return o1.getVolume().compareTo(o2.getVolume());
            }
            case "temperature": {
                return o1.getTemperature().compareTo(o2.getTemperature());
            }
        }
        return 0;


    }
}

File: src/main/java/org/example/HomeWork/hw7/DrinksComparator.py
from Drinks import Drinks

class DrinksComparator(Drinks):
    def init(self, type: str):
        self.type = type
def compare(self, o1: T, o2: T) -> int:
    if self.type == "volume":
        return o1.getVolume() - o2.getVolume()
    elif self.type == "temperature":
        return o1.getTemperature() - o2.getTemperature()
    else:
        return 0